Core.UnitTests/MutableReflection/Implementation/MemberSelectorTest.cs

```diff
@@ -507,69 +526,69 @@ public void SelectSingleEvent_Ambiguous ()
       var bindingFlags = (BindingFlags) 1;
 
       _bindingFlagsEvaluatorMock
-          .Expect (mock => mock.HasRightAttributes (Arg<MethodAttributes>.Is.Anything, Arg.Is(bindingFlags)))
-          .Return (true)
-          .Repeat.Twice();
+          .SetupSequence (mock => mock.HasRightAttributes (It.IsAny<MethodAttributes>(), It.Is<BindingFlags> (param => param == bindingFlags)))
+          .Returns (true)
+          .Returns (true);
 
       _selector.SelectSingleEvent (events, bindingFlags, "Event", _someDeclaringType);
     }
 
     private Type CreateTypeStub (string name = "Unspecified", TypeAttributes attributes = TypeAttributes.Public)
     {
-      var typeStub = MockRepository.GenerateStub<Type>();
-      typeStub.Stub (stub => stub.Name).Return (name);
-      typeStub.Stub (stub => stub.Attributes).Return (attributes);
-
-      return typeStub;
+        var assemblyName = new AssemblyName { Name = name };
+        var assemblyBuilder = Thread.GetDomain().DefineDynamicAssembly(assemblyName, AssemblyBuilderAccess.Run);
+        var moduleBuilder = assemblyBuilder.DefineDynamicModule (assemblyBuilder.GetName().Name, false);
+        var typeBuilder = moduleBuilder.DefineType (name, attributes);
+        return typeBuilder.CreateType();
     }
 
     private FieldInfo CreateFieldStub (
         string name = "Unspecified", FieldAttributes attributes = FieldAttributes.PrivateScope, Type declaringType = null)
     {
-      var fieldStub = MockRepository.GenerateStub<FieldInfo>();
-      fieldStub.Stub (stub => stub.Name).Return (name);
-      fieldStub.Stub (stub => stub.Attributes).Return (attributes);
-      fieldStub.Stub (stub => stub.DeclaringType).Return (declaringType);
+      var fieldStub = new Mock<FieldInfo>();
+      fieldStub.SetupGet (stub => stub.Name).Returns (name);
+      fieldStub.SetupGet (stub => stub.Attributes).Returns (attributes);
+      fieldStub.SetupGet (stub => stub.DeclaringType).Returns (declaringType);
 
-      return fieldStub;
+      return fieldStub.Object;
     }
 
     private ConstructorInfo CreateConstructorStub (MethodAttributes attributes = MethodAttributes.PrivateScope)
     {
-      var constructorStub = MockRepository.GenerateStub<ConstructorInfo>();
-      constructorStub.Stub (stub => stub.Name).Repeat.Never();
-      constructorStub.Stub (stub => stub.Attributes).Return (attributes);
-      return constructorStub;
+      var constructorStub = new Mock<ConstructorInfo>();
+      constructorStub.SetupGet (stub => stub.Name).Throws (new AssertionException ("Do not access the Name property!"));
```

Comment:
I was under the impression that this is the first time we're doing something like that, since its purpose is to emulate RhinoMocks' `Repeat.Never()` behavior

Reply:
Nope, I do that in RhinoMocks all the time since I don't like repeat never (or knew I'd have to give it up and didn't like it since then...)

Reply:
Good news, we have both in trunk :|

![image](https://user-images.githubusercontent.com/24362621/87015366-9fb33600-c1cd-11ea-976b-017ed7be4680.png)

Reply:
Okay. more usages for InvalidOperationException, but only because there was copy&paste involved. please test behavior to make sure it's all good and then update trunk to use AssertionException uniformly.

Reply:
Only the stack trace seems to be different.

InvalidOperationException:

```
System.InvalidOperationException : Do not call me please!
   at Moq.MethodCall.ThrowExceptionResponse.RespondTo(Invocation invocation)
   at Moq.MethodCall.ExecuteCore(Invocation invocation)
...
```

AssertionException:

```
Do not call me please!
   at Moq.MethodCall.ThrowExceptionResponse.RespondTo(Invocation invocation)
   at Moq.MethodCall.ExecuteCore(Invocation invocation)
...
```

Reply:
yeah, ok. that's probably okay to use AssertionEx.